### ISSUE TITLE
Not able to compile on Ubuntu 20.04.1

### ISSUE THREAD

**Issue by gio1993** [mike4192/spotMicro#26]
Hi, 
I've cloned the project int my catkin_ws. I'm running Ubuntu 20.04 and ROS noetic. I've also downloaded the libi2c-dev_3.1.1-1.deb package and installed it. When I try compiling my catkin_ws I get a set of errors due to undefined references. Here's my log:

`giovanni@Giovanni-HP:~/catkin_ws$ catkin_make -DCMAKE_BUILD_TYPE=Release
Base path: /home/giovanni/catkin_ws
Source space: /home/giovanni/catkin_ws/src
Build space: /home/giovanni/catkin_ws/build
Devel space: /home/giovanni/catkin_ws/devel
Install space: /home/giovanni/catkin_ws/install
####
####
-- Using CATKIN_DEVEL_PREFIX: /home/giovanni/catkin_ws/devel
-- Using CMAKE_PREFIX_PATH: /home/giovanni/catkin_ws/devel;/opt/ros/noetic
-- This workspace overlays: /home/giovanni/catkin_ws/devel;/opt/ros/noetic
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.8.5", minimum required is "3") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python3
-- Using Debian Python package layout
-- Using empy: /usr/lib/python3/dist-packages/em.py
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/giovanni/catkin_ws/build/test_results
-- Forcing gtest/gmock from source, though one was otherwise available.
-- Found gtest sources under '/usr/src/googletest': gtests will be built
-- Found gmock sources under '/usr/src/googletest': gmock will be built
-- Found PythonInterp: /usr/bin/python3 (found version "3.8.5") 
-- Using Python nosetests: /usr/bin/nosetests3
-- catkin 0.8.9
-- BUILD_SHARED_LIBS is on
-- BUILD_SHARED_LIBS is on
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- ~~  traversing 11 packages in topological order:
-- ~~  - manipulator_moveit_config
-- ~~  - i2cpwm_board
-- ~~  - lcd_monitor
-- ~~  - servo_move_keyboard
-- ~~  - spot_micro_keyboard_command
-- ~~  - spot_micro_motion_cmd
-- ~~  - spot_micro_plot
-- ~~  - spot_micro_simple_command
-- ~~  - spot_micro_walk
-- ~~  - manipulator_description
-- ~~  - moveit_converter
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- +++ processing catkin package: 'manipulator_moveit_config'
-- ==> add_subdirectory(manipulator_moveit_config)
-- +++ processing catkin package: 'i2cpwm_board'
-- ==> add_subdirectory(spotMicro/ros-i2cpwmboard)
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
-- i2cpwm_board: 6 messages, 4 services
-- +++ processing catkin package: 'lcd_monitor'
-- ==> add_subdirectory(spotMicro/lcd_monitor)
-- +++ processing catkin package: 'servo_move_keyboard'
-- ==> add_subdirectory(spotMicro/servo_move_keyboard)
-- +++ processing catkin package: 'spot_micro_keyboard_command'
-- ==> add_subdirectory(spotMicro/spot_micro_keyboard_command)
-- +++ processing catkin package: 'spot_micro_motion_cmd'
-- ==> add_subdirectory(spotMicro/spot_micro_motion_cmd)
-- +++ processing catkin package: 'spot_micro_plot'
-- ==> add_subdirectory(spotMicro/spot_micro_plot)
-- +++ processing catkin package: 'spot_micro_simple_command'
-- ==> add_subdirectory(spotMicro/spot_micro_simple_command)
-- +++ processing catkin package: 'spot_micro_walk'
-- ==> add_subdirectory(spotMicro/spot_micro_walk)
-- +++ processing catkin package: 'manipulator_description'
-- ==> add_subdirectory(manipulator_description)
-- +++ processing catkin package: 'moveit_converter'
-- ==> add_subdirectory(moveit_converter)
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
-- moveit_converter: 1 messages, 0 services
-- Configuring done
-- Generating done
-- Build files have been written to: /home/giovanni/catkin_ws/build
####
####
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_PositionArray
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_DriveMode
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_StopServos
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_ServoConfigArray
[  0%] Built target std_msgs_generate_messages_cpp
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_ServosConfig
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_Servo
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_Position
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_ServoArray
[  0%] Built target std_msgs_generate_messages_py
[  0%] Built target std_msgs_generate_messages_nodejs
[  0%] Built target std_msgs_generate_messages_eus
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_ServoConfig
Scanning dependencies of target spot_micro_kinematics
[  0%] Built target std_msgs_generate_messages_lisp
[  0%] Built target _i2cpwm_board_generate_messages_check_deps_IntValue
[  2%] Built target gtest
[  2%] Built target sensor_msgs_generate_messages_cpp
[  3%] Building CXX object spotMicro/spot_micro_motion_cmd/libs/spot_micro_kinematics_cpp/CMakeFiles/spot_micro_kinematics.dir/src/spot_micro_kinematics.cpp.o
[  5%] Built target move_group_1
[  5%] Built target sensor_msgs_generate_messages_nodejs
[  6%] Building CXX object spotMicro/spot_micro_motion_cmd/libs/spot_micro_kinematics_cpp/CMakeFiles/spot_micro_kinematics.dir/src/utils.cpp.o
[  6%] Built target sensor_msgs_generate_messages_py
[  6%] Built target sensor_msgs_generate_messages_lisp
[  6%] Built target _moveit_converter_generate_messages_check_deps_ArmJointState
[  6%] Built target sensor_msgs_generate_messages_eus
[  7%] Generating C++ code from i2cpwm_board/Servo.msg
[  8%] Building CXX object spotMicro/spot_micro_motion_cmd/libs/spot_micro_kinematics_cpp/CMakeFiles/spot_micro_kinematics.dir/src/spot_micro_leg.cpp.o
[ 10%] Generating C++ code from i2cpwm_board/ServoArray.msg
[ 11%] Generating C++ code from i2cpwm_board/ServoConfig.msg
[ 12%] Generating C++ code from i2cpwm_board/ServoConfigArray.msg
[ 13%] Generating C++ code from i2cpwm_board/Position.msg
[ 14%] Generating C++ code from i2cpwm_board/PositionArray.msg
[ 15%] Generating C++ code from i2cpwm_board/IntValue.srv
[ 16%] Generating C++ code from i2cpwm_board/ServosConfig.srv
[ 17%] Generating C++ code from i2cpwm_board/DriveMode.srv
[ 18%] Generating C++ code from i2cpwm_board/StopServos.srv
[ 20%] Generating Python from MSG i2cpwm_board/Servo
[ 21%] Generating Javascript code from i2cpwm_board/Servo.msg
[ 22%] Generating Python from MSG i2cpwm_board/ServoArray
[ 23%] Generating Javascript code from i2cpwm_board/ServoArray.msg
[ 23%] Built target i2cpwm_board_generate_messages_cpp
[ 24%] Generating EusLisp code from i2cpwm_board/Servo.msg
[ 25%] Generating Javascript code from i2cpwm_board/ServoConfig.msg
[ 26%] Generating Python from MSG i2cpwm_board/ServoConfig
[ 27%] Generating Javascript code from i2cpwm_board/ServoConfigArray.msg
[ 28%] Generating EusLisp code from i2cpwm_board/ServoArray.msg
[ 30%] Generating Python from MSG i2cpwm_board/ServoConfigArray
[ 31%] Generating Javascript code from i2cpwm_board/Position.msg
[ 32%] Generating EusLisp code from i2cpwm_board/ServoConfig.msg
[ 33%] Generating Python from MSG i2cpwm_board/Position
[ 34%] Generating Javascript code from i2cpwm_board/PositionArray.msg
[ 35%] Generating Javascript code from i2cpwm_board/IntValue.srv
[ 36%] Generating EusLisp code from i2cpwm_board/ServoConfigArray.msg
[ 37%] Generating Python from MSG i2cpwm_board/PositionArray
[ 38%] Generating Javascript code from i2cpwm_board/ServosConfig.srv
[ 41%] Generating Javascript code from i2cpwm_board/DriveMode.srv
[ 41%] Generating Python code from SRV i2cpwm_board/IntValue
[ 42%] Generating EusLisp code from i2cpwm_board/Position.msg
[ 43%] Generating Python code from SRV i2cpwm_board/ServosConfig
[ 44%] Generating Javascript code from i2cpwm_board/StopServos.srv
[ 45%] Generating EusLisp code from i2cpwm_board/PositionArray.msg
[ 45%] Built target i2cpwm_board_generate_messages_nodejs
[ 46%] Generating Python code from SRV i2cpwm_board/DriveMode
[ 47%] Generating Lisp code from i2cpwm_board/Servo.msg
[ 48%] Generating EusLisp code from i2cpwm_board/IntValue.srv
[ 50%] Generating Lisp code from i2cpwm_board/ServoArray.msg
[ 51%] Generating Python code from SRV i2cpwm_board/StopServos
[ 52%] Generating Lisp code from i2cpwm_board/ServoConfig.msg
[ 53%] Generating EusLisp code from i2cpwm_board/ServosConfig.srv
[ 54%] Generating Python msg __init__.py for i2cpwm_board
[ 55%] Generating Lisp code from i2cpwm_board/ServoConfigArray.msg
[ 56%] Generating Lisp code from i2cpwm_board/Position.msg
[ 57%] Generating EusLisp code from i2cpwm_board/DriveMode.srv
[ 58%] Generating Python srv __init__.py for i2cpwm_board
[ 60%] Linking CXX shared library /home/giovanni/catkin_ws/devel/lib/libspot_micro_kinematics.so
[ 61%] Generating Lisp code from i2cpwm_board/PositionArray.msg
[ 61%] Built target i2cpwm_board_generate_messages_py
[ 62%] Generating EusLisp code from i2cpwm_board/StopServos.srv
[ 63%] Generating Lisp code from i2cpwm_board/IntValue.srv
[ 65%] Built target gtest_main
[ 66%] Built target moveit_converter_generate_messages_cpp
[ 67%] Generating Lisp code from i2cpwm_board/ServosConfig.srv
[ 68%] Built target moveit_converter_generate_messages_nodejs
[ 68%] Built target spot_micro_kinematics
[ 70%] Built target i2cpwm_board_generate_messages_eus
[ 72%] Built target moveit_converter_generate_messages_py
[ 73%] Built target moveit_converter_generate_messages_lisp
[ 74%] Generating Lisp code from i2cpwm_board/DriveMode.srv
[ 75%] Generating Lisp code from i2cpwm_board/StopServos.srv
[ 77%] Built target moveit_converter_generate_messages_eus
Scanning dependencies of target i2cpwm_board
Scanning dependencies of target smfsm
[ 78%] Building CXX object spotMicro/ros-i2cpwmboard/CMakeFiles/i2cpwm_board.dir/src/i2cpwm_controller.cpp.o
Scanning dependencies of target test_spot_micro_leg
[ 78%] Built target i2cpwm_board_generate_messages_lisp
[ 80%] Building CXX object spotMicro/spot_micro_motion_cmd/CMakeFiles/smfsm.dir/src/smfsm/spot_micro_state.cpp.o
[ 81%] Building CXX object spotMicro/spot_micro_motion_cmd/libs/spot_micro_kinematics_cpp/CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o
[ 82%] Building CXX object spotMicro/spot_micro_motion_cmd/CMakeFiles/smfsm.dir/src/smfsm/spot_micro_idle.cpp.o
[ 83%] Linking CXX executable test_spot_micro_leg
/home/giovanni/catkin_ws/src/spotMicro/ros-i2cpwmboard/src/i2cpwm_controller.cpp: In function ‘int _load_params()’:
/home/giovanni/catkin_ws/src/spotMicro/ros-i2cpwmboard/src/i2cpwm_controller.cpp:1567:1: warning: no return statement in function returning non-void [-Wreturn-type]
 1567 | }
      | ^
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_global_coordinates_Test::TestBody()':
test_spot_micro_leg.cpp:(.text+0x3a9): undefined reference to `testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x41b): undefined reference to `testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x48d): undefined reference to `testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x4f1): undefined reference to `testing::Message::Message()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x521): undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x52c): undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x534): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x561): undefined reference to `testing::Message::Message()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x591): undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x59c): undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x5a4): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x5d1): undefined reference to `testing::Message::Message()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x601): undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x60c): undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0x614): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_local_coordinates_Test::TestBody()':
test_spot_micro_leg.cpp:(.text+0x9d6): undefined reference to `testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xa7a): undefined reference to `testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xb1d): undefined reference to `testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xb81): undefined reference to `testing::Message::Message()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xbb3): undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xbbe): undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xbc6): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xbf1): undefined reference to `testing::Message::Message()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xc23): undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xc2e): undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xc36): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xc61): undefined reference to `testing::Message::Message()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xc93): undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xc9e): undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text+0xca6): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_global_coordinates_Test::~setFootPos_global_coordinates_Test()':
test_spot_micro_leg.cpp:(.text._ZN34setFootPos_global_coordinates_TestD2Ev[_ZN34setFootPos_global_coordinates_TestD5Ev]+0xf): undefined reference to `testing::Test::~Test()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_global_coordinates_Test::~setFootPos_global_coordinates_Test()':
test_spot_micro_leg.cpp:(.text._ZN34setFootPos_global_coordinates_TestD0Ev[_ZN34setFootPos_global_coordinates_TestD5Ev]+0x13): undefined reference to `testing::Test::~Test()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_local_coordinates_Test::~setFootPos_local_coordinates_Test()':
test_spot_micro_leg.cpp:(.text._ZN33setFootPos_local_coordinates_TestD2Ev[_ZN33setFootPos_local_coordinates_TestD5Ev]+0xf): undefined reference to `testing::Test::~Test()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_local_coordinates_Test::~setFootPos_local_coordinates_Test()':
test_spot_micro_leg.cpp:(.text._ZN33setFootPos_local_coordinates_TestD0Ev[_ZN33setFootPos_local_coordinates_TestD5Ev]+0x13): undefined reference to `testing::Test::~Test()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `testing::internal::TestFactoryImpl<setFootPos_local_coordinates_Test>::CreateTest()':
test_spot_micro_leg.cpp:(.text._ZN7testing8internal15TestFactoryImplI33setFootPos_local_coordinates_TestE10CreateTestEv[_ZN7testing8internal15TestFactoryImplI33setFootPos_local_coordinates_TestE10CreateTestEv]+0x1c): undefined reference to `testing::Test::Test()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `testing::internal::TestFactoryImpl<setFootPos_global_coordinates_Test>::CreateTest()':
test_spot_micro_leg.cpp:(.text._ZN7testing8internal15TestFactoryImplI34setFootPos_global_coordinates_TestE10CreateTestEv[_ZN7testing8internal15TestFactoryImplI34setFootPos_global_coordinates_TestE10CreateTestEv]+0x1c): undefined reference to `testing::Test::Test()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_global_coordinates_Test::TestBody() [clone .cold]':
test_spot_micro_leg.cpp:(.text.unlikely+0x4): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text.unlikely+0x33): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text.unlikely+0x62): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `setFootPos_local_coordinates_Test::TestBody() [clone .cold]':
test_spot_micro_leg.cpp:(.text.unlikely+0x92): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text.unlikely+0xc1): undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o:test_spot_micro_leg.cpp:(.text.unlikely+0xf0): more undefined references to `testing::internal::AssertHelper::~AssertHelper()' follow
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `testing::internal::SuiteApiResolver<testing::Test>::GetSetUpCaseOrSuite(char const*, int)':
test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci]+0x28): undefined reference to `testing::internal::IsTrue(bool)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci]+0x6a): undefined reference to `testing::internal::GTestLog::GTestLog(testing::internal::GTestLogSeverity, char const*, int)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci]+0xe7): undefined reference to `testing::internal::GTestLog::~GTestLog()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE19GetSetUpCaseOrSuiteEPKci]+0x11f): undefined reference to `testing::internal::GTestLog::~GTestLog()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `testing::internal::SuiteApiResolver<testing::Test>::GetTearDownCaseOrSuite(char const*, int)':
test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci]+0x28): undefined reference to `testing::internal::IsTrue(bool)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci]+0x6a): undefined reference to `testing::internal::GTestLog::GTestLog(testing::internal::GTestLogSeverity, char const*, int)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci]+0xe7): undefined reference to `testing::internal::GTestLog::~GTestLog()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text._ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci[_ZN7testing8internal16SuiteApiResolverINS_4TestEE22GetTearDownCaseOrSuiteEPKci]+0x11f): undefined reference to `testing::internal::GTestLog::~GTestLog()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o: in function `_GLOBAL__sub_I__ZN33setFootPos_local_coordinates_Test10test_info_E':
test_spot_micro_leg.cpp:(.text.startup+0x97): undefined reference to `testing::internal::GetTestTypeId()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text.startup+0xfa): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text.startup+0x172): undefined reference to `testing::internal::GetTestTypeId()'
/usr/bin/ld: test_spot_micro_leg.cpp:(.text.startup+0x1d1): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o:(.data.rel.ro._ZTI33setFootPos_local_coordinates_Test[_ZTI33setFootPos_local_coordinates_Test]+0x10): undefined reference to `typeinfo for testing::Test'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o:(.data.rel.ro._ZTI34setFootPos_global_coordinates_Test[_ZTI34setFootPos_global_coordinates_Test]+0x10): undefined reference to `typeinfo for testing::Test'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o:(.data.rel.ro._ZTV33setFootPos_local_coordinates_Test[_ZTV33setFootPos_local_coordinates_Test]+0x20): undefined reference to `testing::Test::SetUp()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o:(.data.rel.ro._ZTV33setFootPos_local_coordinates_Test[_ZTV33setFootPos_local_coordinates_Test]+0x28): undefined reference to `testing::Test::TearDown()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o:(.data.rel.ro._ZTV34setFootPos_global_coordinates_Test[_ZTV34setFootPos_global_coordinates_Test]+0x20): undefined reference to `testing::Test::SetUp()'
/usr/bin/ld: CMakeFiles/test_spot_micro_leg.dir/tests/test_spot_micro_leg.cpp.o:(.data.rel.ro._ZTV34setFootPos_global_coordinates_Test[_ZTV34setFootPos_global_coordinates_Test]+0x28): undefined reference to `testing::Test::TearDown()'
/usr/bin/ld: ../../../../lib/libgtest_main.so: undefined reference to `testing::UnitTest::GetInstance()'
/usr/bin/ld: ../../../../lib/libgtest_main.so: undefined reference to `testing::UnitTest::Run()'
/usr/bin/ld: ../../../../lib/libgtest_main.so: undefined reference to `testing::InitGoogleTest(int*, char**)'
collect2: error: ld returned 1 exit status
make[2]: *** [spotMicro/spot_micro_motion_cmd/libs/spot_micro_kinematics_cpp/CMakeFiles/test_spot_micro_leg.dir/build.make:87: spotMicro/spot_micro_motion_cmd/libs/spot_micro_kinematics_cpp/test_spot_micro_leg] Errore 1
make[1]: *** [CMakeFiles/Makefile2:2042: spotMicro/spot_micro_motion_cmd/libs/spot_micro_kinematics_cpp/CMakeFiles/test_spot_micro_leg.dir/all] Errore 2
make[1]: *** Attesa per i processi non terminati....
[ 84%] Building CXX object spotMicro/spot_micro_motion_cmd/CMakeFiles/smfsm.dir/src/smfsm/spot_micro_stand.cpp.o
[ 86%] Building CXX object spotMicro/spot_micro_motion_cmd/CMakeFiles/smfsm.dir/src/smfsm/spot_micro_transition_idle.cpp.o
[ 86%] Building CXX object spotMicro/spot_micro_motion_cmd/CMakeFiles/smfsm.dir/src/smfsm/spot_micro_transition_stand.cpp.o
[ 87%] Building CXX object spotMicro/spot_micro_motion_cmd/CMakeFiles/smfsm.dir/src/smfsm/spot_micro_walk.cpp.o
[ 88%] Linking CXX executable /home/giovanni/catkin_ws/devel/lib/i2cpwm_board/i2cpwm_board
[ 88%] Built target i2cpwm_board
[ 90%] Linking CXX shared library /home/giovanni/catkin_ws/devel/lib/libsmfsm.so
[ 90%] Built target smfsm
make: *** [Makefile:141: all] Errore 2
Invoking "make -j4 -l4" failed
`
Any idea of what the issue can it be?
Regards.

**Comment by mike4192** [comment]
Unfortunately ROS Kinetic, which this project is based on, is only supported in Ubuntu 16.04, so not sure my suggestions will help at all, but it looks like you may be missing gtest.

Some random suggestions:

- Make sure you have gtest installed: `sudo apt-get install libgtest-dev`
- Make sure you initialized all the submodules: 
```
git submodule update --init --recursive
git submodule update --recursive
```

**Comment by gio1993** [comment]
Thank you for your suggestion but libgtest-dev was already installed and updated.
I reinstalled ROS Noetic, followed your instructions and after some real struggling with python versions I got it compiling and running (on an old Laptop for now). During next days I'll implement it on RPI4 and I'll try it on Ubuntu 20.1. In case i'll not be able to make it run, I'll try the 20.04 release. 
I'll let you know if you are interested.
Regards.
